lib/internal/test_runner/mock/mock_timers.js

```diff
@@ -292,31 +383,93 @@ class MockTimers {
     }
   }
 
-  enable(timers = SUPPORTED_TIMERS) {
+  enable(...args) {
     if (this.#isEnabled) {
-      throw new ERR_INVALID_STATE(
-        'MockTimers is already enabled!',
-      );
+      throw new ERR_INVALID_STATE('MockTimers is already enabled!');
+    }
+
+    if (args.length <= 0) {
+      this.#timersInContext = SUPPORTED_TIMERS;
+      this.#now = 0;
+      this.#toggleEnableTimers(true);
+      return;
     }
 
-    validateArray(timers, 'timers');
+    let timersDefined = false;
+    // Checks if the first argument is the timer array or the initial time
+    if (ArrayIsArray(args[0])) {
+      validateArray(args[0], 'timers');
+      // Check that the timers passed are supported
+      ArrayPrototypeForEach(args[0], (timer) => {
+        if (!ArrayPrototypeIncludes(SUPPORTED_TIMERS, timer)) {
+          throw new ERR_INVALID_ARG_VALUE(
+            'timers',
+            timer,
+            `option ${timer} is not supported`,
+          );
+        }
+      });
+
+      this.#timersInContext = args[0];
+      timersDefined = true;
+      this.#now = 0;
+    } else if (this.#isValidDateWithGetTime(args[0])) {
+      // First argument is the initial time as Date
+      this.#now = args[0].getTime();
+      this.#timersInContext = SUPPORTED_TIMERS;
+    } else if (validateNumber(args[0], 'initialTime') === undefined) {
+      // First argument is the initial time as number
+      this.#assertTimeArg(args[0]);
+      this.#now = args[0];
+      this.#timersInContext = SUPPORTED_TIMERS;
+    } else {
+      throw ERR_INVALID_ARG_VALUE(
+        'enableTimers',
+        args[0],
+        'the first argument must be an array of strings, a Date instance or a positive integer',
+      );
+    }
 
-    // Check that the timers passed are supported
-    ArrayPrototypeForEach(timers, (timer) => {
-      if (!ArrayPrototypeIncludes(SUPPORTED_TIMERS, timer)) {
-        throw new ERR_INVALID_ARG_VALUE(
-          'timers',
-          timer,
-          `option ${timer} is not supported`,
+    // Checks if the second argument is the initial time
+    if (args[1] && timersDefined) {
+      if (this.#isValidDateWithGetTime(args[1])) {
+        this.#now = args[1].getTime();
+      } else if (validateNumber(args[1], 'initialTime') === undefined) {
+        this.#assertTimeArg(args[1]);
+        this.#now = args[1];
+      } else {
+        throw ERR_INVALID_ARG_VALUE(
+          'initialEpoch',
+          args[1],
+          'initial epoch must be a Date instance or a positive integer',
         );
       }
-    });
+    }
 
-    this.#timersInContext = timers;
-    this.#now = DateNow();
     this.#toggleEnableTimers(true);
   }
 
+  /**
+   * @param {number | Date} time The epoch to set the current time to.
+   */
+  setTime(time = 0) {
+    let assertedTime = time;
+    if (this.#isValidDateWithGetTime(time)) {
+      assertedTime = time.getTime();
```

Comment:
and here

Reply:
Same idea too, all the places where we can receive the timestamp can receive a Date object. But this place wouldn't make sense since the date object can be mocked already, so maybe here it's better to rip it off

Reply:
right but we shouldn't rely on `getTime` existing at runtime; that's what primordials are for